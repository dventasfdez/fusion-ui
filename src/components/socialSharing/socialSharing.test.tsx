import React from 'react';
import renderer from 'react-test-renderer';
import {render} from '@testing-library/react';
import SocialSharing from './socialSharing';

const socialSharingExample = () => (
  <div className=.stepone-ui">
    <SocialSharing title="Copy link" />
  </div>
);

it('Social sharing module should render', () => {
  const component = renderer.create(socialSharingExample());
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('Display social media icon', () => {
  const {container} = render(socialSharingExample());

  const socialMediaItem = container.getElementsByClassName('social-sharing-icon linkedin')[0];
  expect(socialMediaItem).toBeInTheDocument();
});

it('Display sharing link', () => {
  const {getByText} = render(socialSharingExample());

  const sharingLink = getByText('Copy link');
  expect(sharingLink).toBeInTheDocument();
});
