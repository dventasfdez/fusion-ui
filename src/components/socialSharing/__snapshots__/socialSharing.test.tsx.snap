// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social sharing module should render 1`] = `
<div
  className="fusion-ui"
>
  <div
    className="social-sharing"
  >
    <a
      className="social-sharing-icon linkedin"
      href="https://www.linkedin.com/shareArticle/?mini=true&url=false"
      target="self"
    />
    <a
      className="social-sharing-icon twitter"
      href="https://twitter.com/intent/tweet?text=false"
      target="self"
    />
    <a
      className="social-sharing-icon facebook"
      href="https://www.facebook.com/sharer/sharer.php?u=false"
      target="self"
    />
    <a
      className="social-sharing-icon whatsapp"
      href="whatsapp://send?text=false"
      target="self"
    />
    <div
      className="link_small"
      onClick={[Function]}
    >
      Copy link
      <span
        className="material-icons right"
      >
        link
      </span>
    </div>
  </div>
</div>
`;
